packages/DrillDownTable/src/components/Pagination/index.tsx

```diff
@@ -45,15 +45,27 @@ function Pagination<T extends object = Dictionary>(props: PaginationProps<T>) {
     pageSizeCategories
   } = props;
 
+  const [pageNumber, setPageNumber] = useState<string>('');
+
+  useEffect(() => {
+    setPageNumber(`${pageIndex + 1}`);
+  }, [pageIndex]);
+
   const onChangePageSize = (e: React.ChangeEvent<HTMLSelectElement>) => {
     setPageSize(Number(e.target.value));
   };
   const onClickPrevious = () => {
     previousPage();
   };
   const onChangePageIndex = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const page = e.target.value ? Number(e.target.value) - 1 : 0;
-    gotoPage(page);
+    setPageNumber(e.target.value);
+
+    if (e.target.value) {
+      const value = Number(e.target.value);
+      const index = value ? (pageOptions.indexOf(value - 1) >= 0 ? value - 1 : 0) : 0;
+      gotoPage(index);
+      setPageNumber(`${index + 1}`);
+    }
```

Comment:
can we have a single `setPageNumber` call here.

Reply:
well not exactly a single call but seem like the value is initially set and then there are more steps to validate it and reset it again